src/propmachinery.jl

```diff
@@ -1,32 +1,45 @@
-import Base: start, next, done
-
 abstract QuPropagatorMethod
 
-immutable QuPropagator{QPM<:QuPropagatorMethod}
-    hamiltonian
-    init_state::QuVector
+immutable QuPropagator{QPM<:QuPropagatorMethod, QVM<:Union(QuBase.AbstractQuVector,QuBase.AbstractQuMatrix)}
+    eq::QuEquation
```

Comment:
It is actually better to use a type parameter (as you did before). Using abstract types for fields is discouraged.